diff --git a/test/src/test_porting_abstraction_layer_operating_system.h b/test/src/test_porting_abstraction_layer_operating_system.h
index 1c0ab03..d387752 100644
--- a/test/src/test_porting_abstraction_layer_operating_system.h
+++ b/test/src/test_porting_abstraction_layer_operating_system.h
@@ -67,6 +67,27 @@
 
   #define TEST_PAL_CALLING_CONVENTION  
 
+  #if( defined ANDROID && !(defined CPU_SET) )
+    #define CPU_SETSIZE 1024
+    #define __NCPUBITS  (8 * sizeof (unsigned long))
+
+    typedef struct {
+       unsigned long __bits[CPU_SETSIZE / __NCPUBITS];
+    } cpu_set_t;
+
+    #define CPU_SET(cpu, cpusetp) \
+      ((cpusetp)->__bits[(cpu)/__NCPUBITS] |= (1UL << ((cpu) % __NCPUBITS)))
+
+    #define CPU_ZERO(cpusetp) \
+      memset((cpusetp), 0, sizeof(cpu_set_t))
+
+    #define SYS_gettid __NR_gettid
+    static inline int sched_setaffinity(pid_t pid, size_t cpusetsize, const cpu_set_t *set)
+    {
+      return syscall(__NR_sched_setaffinity, pid, cpusetsize, set);
+    }
+  #endif
+
 #endif

